### PR TITLE
Ana/adding checks. tentative

### LINKED ISSUE #3496: Refactor all update fields in SmartQueries
All `update` functions inside SmartQuery should now add a check to see if the data was succesfully fetched:

```
      update(data) {
        if (!data.overview) {
          return {
            totalRewards: 0
          }
        }
        /* istanbul ignore next */
        return {
          ...data.overview,
          totalRewards: Number(data.overview.totalRewards)
        }
      }
```

In case it wasn't, we return either "", 0 or {} depending on the case.

As I understood it from the meeting we just had, this should now be applied to <b>every</b> SmartQuery, doesn't matter if it has nested values or not.

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #3496 

**Description:**

In this PR I aim to refactor all update functions inside Smartqueries so we prevent all the errors when a query fails.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by jbibla** [comment]
nice! defensive programming 🛡 

tests are failing...

**Comment by Bitcoinera** [comment]
> nice! defensive programming 
> 
> tests are failing...

Yes, thank you. I will first see if I am going the right way :pray:

File: src/components/network/PageBlock.vue [hunk 1]
```diff
@@ -104,7 +104,13 @@ export default {
           networkId: this.network
         }
       },
+      /* istanbul ignore next */
       update: function(result) {
+        if (!result) {
```
Comment (faboweb) [inline src/components/network/PageBlock.vue]: this is wrong right? shouldn't this return just an array as the good path is returning `result.validators`

Comment (Bitcoinera) [reply src/components/network/PageBlock.vue]: yes, looks quite wrong... :eyes:  thx

Comment (Bitcoinera) [reply src/components/network/PageBlock.vue]: I think we better wait and see if we adopt @iambeone 's solution.

He has being demoing it to me earlier and it really does look far superior to this one.

I am tempted to close this PR quite frankly.

**Comment by Bitcoinera** [comment]
I close this as I think Aleksei's solution is by far better than this one

**Pull request closed without merging**